presto-docs/src/main/sphinx/admin/function-namespace-managers.rst

```diff
@@ -0,0 +1,76 @@
+===========================
+Function Namespace Managers
+===========================
+
+.. warning::
+
+    This is an experimental feature being actively developed. The way
+    Function Namespace Managers are configured might be changed.
+
+Function namespace managers support storing and retrieving user-defined
+functions with a persistent storage, allowing users to create, alter,
+and delete functions, as well as to reference them within queries.
+
+Function Namespace
+------------------
+
+A function namespace is a catalog-schema name (i.e. ``example.test``)
+in which each function can be uniquely identified by its signature,
+that is, its name and parameter type list.
+
+The function namespace, such as ``example.test``, is not a full-fledged
+schema as it does not support storing tables and views, but only
+functions.
+
+Each instance of function namespace manager manages a catalog. Creating
+and managing schema in those catalogs are not supported.
+
+Configuration
+-------------
+
+Presto provides a MySQL-based function namespace manager. It is
+pluggable and can be configured by an administrator.
+
+To instantiate a MySQL-based function namespace manager that manages
+catalog ``example``, administrator needs to first have a running MySQL
+server. Suppose the MySQL server can be reached at ``localhost:1080``,
+add a file ``etc/function-namespace/example.properties`` with the
+following contents::
+
+    function-namespace-manager.name=mysql
+    database-url=localhost:1080
+    function-namespaces-table-name=example_function_namespaces
+    functions-table-name=example_sql_functions
+
+When Presto first starts with the above MySQL function namespace
+manager configuration, two MySQL tables will be created if they do
+not exist.
+
+- ``example_function_namespaces`` stores function namespaces of
+  the catalog ``example``.
+- ``example_sql_functions`` stores SQL-invoked functions of the
+  catalog ``example``.
+
+If multiple function namespace managers are instantiated, they may use
+the same set of tables, or different tables / MySQL databases.
+
+To create a new function namespace, insert into the
```

Comment:
I'm a bit confused about this.  
1. If each function namespace manager only manages a single catalog, then why does the function_namespaces catalog take in both a catalog and schema name. 
2. Above you said you can't add schemas

Reply:
For 1: Different function namespace managers may use the same `function_namespaces`, and hence the table may contain function namespaces not managed by a single function namespace manager.

For 2:
a. You cannot create a function namespace with `CREATE SCHEMA`.
b. A function namespace is not exactly the same as a schema, i.e., you cannot create tables and views in it.
Although I agree it might just be ok to call it a schema when we support (a).

Reply:
2 makes sense.  

For 1 are you saying that different function namespace managers may use the same table to keep track of their namespaces?  It would be helpful to state that explicitly somewhere.   Also, do you need to have separate configuration files for each catalog, or is it enough to stick new info in the function_namespaces table?

Reply:
The section above states it:
```
If multiple function namespace managers are instantiated, they may use
the same set of tables, or different tables / MySQL databases.
```

Reply:
I expanded the explanation for multiple function namespace manager.
```
Multiple function namespace managers can be instantiated by placing
configuration properties file under ``etc/function-namespace``. They
may use the same tables, or different tables / MySQL databases. If the
same MySQL table are used by multiple function namespace managers,
each manager will only create and interact with entries of the catalog
to which it binds.
```